vars/piperExecuteBin.groovy

```diff
@@ -45,6 +47,14 @@ void call(Map parameters = [:], String stepName, String metadataFile, List crede
                 echo "Context Config: ${config}"
             }
 
+            // prepare stashes
+            // first eliminate empty stahes
+            config.stashContent = utils.unstashAll(config.stashContent)
+            // then make sure that commonPipelineEnvironment, config, ... is also available when step stashing is active
+            if (config.stashContent?.size() > 0) {
```

Comment:
Even from the comments I don't get what is happening here. E.g. Why ">0"? Why unstashing before at all? Doesn't this cost an unnecessary amount of time?

Reply:
There is no other way to identify if stashes are filled than to `unstash`
This is a typical we have been using in all relevant `groovy`-based steps, e.g.:
https://github.com/SAP/jenkins-library/blob/master/vars/gaugeExecuteTests.groovy#L128

The `>0` ensures that also cases are handled where `stashContent` is defined (e.g. via defaults) but there is no content in the stashes.
In such a case (`>0`>) we need to make sure that also the stash `pipelineConfigAndTests` is considered since the stashes take precedence over the complete workspace and we otherwise cannot guarantee that the files (config, ...) are available.

Now, if we do not check if any of the stashes is filled (case `==0`) we will add `pipelineConfigAndTests` and in a Kubernetes environment we will break all usages since now the step only has this stash but nothing which is stored in the workspace ...
see https://github.com/SAP/jenkins-library/blob/master/vars/dockerExecuteOnKubernetes.groovy#L255